tpm2tools/signer.go

```diff
@@ -0,0 +1,77 @@
+package tpm2tools
+
+import (
+	"crypto"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/google/go-tpm/tpm2"
+)
+
+// Global mutex to protect against concurrent TPM access.
+var signerMutex sync.Mutex
+
+type tpmSigner struct {
+	Key  *Key
+	Hash crypto.Hash
+}
+
+// Public returns the tpmSigners public key.
+func (signer *tpmSigner) Public() crypto.PublicKey {
+	return signer.Key.PublicKey()
+}
+
+// Sign uses the TPM key to  sign the digest.
+// The digest must be hashed from the same hash algorithm as the keys scheme.
+// The opts hash function must also match the keys scheme.
+// Concurrent use of Sign is thread safe, but it is not safe to access the TPM
+// from other sources while Sign is executing.
+func (signer *tpmSigner) Sign(_ io.Reader, digest []byte, opts crypto.SignerOpts) (signature []byte, err error) {
```

Comment:
For consistency with https://golang.org/pkg/crypto/rsa/#PrivateKey.Sign, we should make sure `opts` isn't a `*PSSOptions`.